### DISCUSSION TITLE
IAM Role for service account not working from containerized spring boot application

### DISCUSSION THREAD

**Discussion by vinaykumar-git6** [aws/aws-sdk-java#2863] [Q&A]
Hi,

My pods running in EKS are not able to assume IAM role with web identity token injected in pod. Below token file has been mounted on my pod.

      AWS_STS_REGIONAL_ENDPOINTS:   regional
      AWS_DEFAULT_REGION:           me-south-1
      AWS_REGION:                   me-south-1
      AWS_ROLE_ARN:                 arn:aws:iam::XXXXXXXXXXX:role/rate-information-cbch-role
      AWS_WEB_IDENTITY_TOKEN_FILE:  /var/run/secrets/eks.amazonaws.com/serviceaccount/token

However my pod is still using the node IAM role. If I remove permission from IAM role pod starts failing. Ideally it should have used IAM role for service account.

**Comment by debora-ito** [comment]
What is the question?
